cli/pkg/kctrl/local/sources/fetch.go

```diff
@@ -0,0 +1,139 @@
+// Copyright 2022 VMware, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+package sources
+
+import (
+	"fmt"
+
+	"github.com/cppforlife/go-cli-ui/ui"
+	cmdcore "github.com/vmware-tanzu/carvel-kapp-controller/cli/pkg/kctrl/cmd/core"
+	buildconfigs "github.com/vmware-tanzu/carvel-kapp-controller/cli/pkg/kctrl/local/buildconfigs"
+	vendirconf "github.com/vmware-tanzu/carvel-vendir/pkg/vendir/config"
+)
+
+const (
+	VendirSyncDirectory = "upstream"
+	vendirFileName      = "vendir.yml"
+	includeAllFiles     = "*"
+)
+
+const (
+	FetchFromGithubRelease       string = "Github Release"
+	FetchFromGit                 string = "Git Repository"
+	FetchFromHelmRepo            string = "Helm Chart from Helm Repository"
+	FetchChartFromGit            string = "Helm Chart from Git Repository"
+	FetchFromLocalDirectory      string = "Local Directory"
+	MultipleFetchOptionsSelected string = "MultipleFetchOptionsSelected"
+)
+
+type FetchConfiguration struct {
```

Comment:
Definitely better, 
I think the returned values make sense in this context as well

Reply:
Although something to think about here is that the package name is sources, so outside this function it would be `sources.Source` 🤔

Reply:
Also renaming GitConfiguration to GitSource, and the others too along similar lines

Reply:
Yep, was about to add that as a comment 👍🏻

Reply:
>Although something to think about here is that the package name is sources, so outside this function it would be sources.Source 🤔

I think that is fine (Example: `resources.Resource`)